pallene/checker.lua

```diff
@@ -145,15 +138,22 @@ end
 --
 
 type_check = function(prog_ast)
-    local ok, err = xpcall(check_program, debug.traceback, prog_ast)
+    local co = coroutine.create(function()
+        check_program(prog_ast)
+    end)
+    local ok, err_msg = coroutine.resume(co)
     if ok then
-        return prog_ast, {} -- TODO:  no {}
-    else
-        if getmetatable(err) == TypeError then
-            return false, { err.msg } -- TODO  no {}
+        if coroutine.status(co) == "dead" then
+            -- User's program passed type checker
+            return prog_ast, {} -- TODO:  no {}
```

Comment:
I'd rather remove this TODO. This is non-informative.

Reply:
Agreed, the issue tracker is a better place for this kind of thing.